external-crates/move/crates/move-compiler/src/cfgir/translate.rs

```diff
@@ -652,12 +673,36 @@ fn function_body(
                     &UniqueMap::new(),
                     &mut cfg,
                 );
+                if context
+                    .env
+                    .supports_feature(context.current_package, FeatureGate::Move2024Optimizations)
+                {
+                    locals = cfgir::optimize::coalesce_locals::optimize(
+                        signature,
+                        &infinite_loop_starts,
+                        locals,
+                        &mut cfg,
+                    );
```

Comment:
Should we move this logic inside of `cfgir::optimize`?

Reply:
Feels like slightly better encapsulation

Reply:
I did it this way to avoid passing the loop starts, but I agree placing it in there does make more sense.